### ISSUE TITLE
GraphQL Complexity Analysis argument handling needs to be improved

### ISSUE THREAD

**Issue by fazlan-nazeem** [wso2/carbon-apimgt#9294]
### Description:

Currently the graphQL const analysis handler assumes that the passed argument can be casted to an Int.  This means that if we pass in an ENUM or a String as an argument, the API call fails in the runtime. e.g

```
query{
  allLifts (status: OPEN){
    name
  }
}

```
Furthermore, it assumes that the passed argument is of type "first" and "last" and takes any int value passed to an argument to the cost analysis. This needs to be handled in a proper way by introducing some kind of a convention because there could be other types of arguments that aren't used for pagination.

### Affected Product Version:
APIM 3.2.0

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---

**Comment by fazlan-nazeem** [comment]
@HiranyaKavishani

**Comment by HiranyaKavishani** [comment]
@fazlan-nazeem Is it always trying to cast the value of the argument into int? If we pass it as ENUM or String, we won't fail the API call, just skip from the place where it takes to consider. So, We shouldn't rely on the other types of attributes except for int values

**Comment by fazlan-nazeem** [comment]
@HiranyaKavishani yes, currently it tries to cast whatever argument value passed. Even if it is an int, we shouldn't cast all arguments as an ID also can be an Int. We should cast only arguments such as "first" and "last" as these are the arguments that are used for slicing(pagination). However, these names are not a convention as well. So ideally a schema could use some other name for these slicing arguments. This[1] describes a specification from Relay which is the connection pattern. We may have to consider these and improve argument handling properly.

[1] https://graphql.org/learn/pagination/
